File: backend/src/main/scala/com/skewwhiffy/auraltester/scales/IntervalNotes.scala
package com.skewwhiffy.auraltester.scales

import com.skewwhiffy.auraltester.notes.AbsoluteNote
import com.skewwhiffy.auraltester.notes.Interval.Interval

class IntervalNotes (
  val bottomNote: AbsoluteNote,
  val interval: Interval
) extends NoteSequence {
  def notes: List[AbsoluteNote] = List(bottomNote, bottomNote + interval)
}

File: backend.kotlin/src/main/kotlin/com/skewwhiffy/notation/model/note/IntervalNotes.kt
package com.skewwhiffy.notation.model.note

import com.skewwhiffy.notation.model.interval.Interval

data class IntervalNotes (
  val bottomNote: AbsoluteNote,
  val interval: Interval
) : NoteSequence {
  override val notes: List<AbsoluteNote> = listOf(bottomNote, bottomNote + interval)
}
